_source/_posts/2022-02-23-angular-async-config.md

```diff
@@ -0,0 +1,340 @@
+---
+layout: blog_post
+title: "Three ways to configure modules in your Angular app"
+author: alisa-duncan
+by: advocate
+communities: [javascript]
+description: "Learn how to add configuration to modules in Angular three different ways - from static, changing configs at build time, and to loading configurations at run-time."
+tags: [javascript, typescript, angular]
+tweets:
+- "How do you handle your module configuration in #Angular apps? 🤔 We cover 3 ways to configure modules and when to use each method."
+- "Needing to load your Okta config from an API in your #Angular app? Learn when to use run-time loading and how to add it to your Okta integration."
+image: blog/angular-async-config/angular-async-config-social.jpg
+type: awareness
+github: https://github.com/oktadev/okta-angular-async-load-example
+---
+
+Configurations are a part of a developer's life. Configuration data is information your app needs to run and may include tokens for third-party systems or settings you pass into libraries. There are different ways to load configuration data as part of application initialization in [Angular](https://angular.io/). Your requirements for configuration data might change based on needs. For example, you may have one unchanging configuration for your app, or you may need a different configuration based on the environment it runs on. We'll cover a few different ways to load configuration values and identify when you should use each method.
+
+> We're covering specific use cases using Angular, so this post assumes you have some experience developing Angular apps. If you're interested in learning more about building your first Angular app, check out the [Angular Getting Started](https://angular.io/guide/what-is-angular) docs, or check out the [links to tutorials](#learn-more) that walk you through integrating Okta into Angular apps.
+
+In this post, we'll cover the following forms of configuration
+* defining configuration directly in code
+* defining configuration for different environments
+* loading configuration data via an API call
+
+We'll show examples, including how to integrate with Okta, for each method. We'll also identify when to use each technique and what to watch out for.
```

Comment:
```suggestion
We'll show examples, including how to integrate with Okta, for each method. Also, we'll identify when to use each technique and what to watch out for.
```
breaks up the repetition of "we'll" to open consecutive sentences. the variety makes for an easier read.